### PR TITLE
Add message & metrics in transaction update

### PR DESCRIPTION
**Pull request by DanielMwendwa**

### CODE REVIEW

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 1]
```diff
@@ -756,35 +756,44 @@ def add_message_to_dataset(self, dataset_id, message, max_segment_size=MAX_SEGME
...
         :param message: The message to be added.
         :type message: core_data_modules.data_models.message.Message
-        :param max_segment_size: the maximum size for a segment, defaults to 2500
+        :param max_segment_size: The maximum size for a segment, defaults to 2500
         :type max_segment_size: int, optional
         """
-        message_exists = self.get_dataset_message(dataset_id, message.message_id) is not None
-        assert not message_exists, f"message with id {message.message_id} already exists."
+        @firestore.transactional
+        def add_in_transaction(transaction, message=message):
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:763]: Do you need to have message here? Why not 
```suggestion
        def add_in_transaction(transaction):
```

File: coda_v2_python_client/firebase_client_wrapper.py [hunk 2]
```diff
@@ -756,35 +756,44 @@ def add_message_to_dataset(self, dataset_id, message, max_segment_size=MAX_SEGME
...
+            message_id = message.message_id
```
Comment (as2388) [inline coda_v2_python_client/firebase_client_wrapper.py:764]: Can you insert a log along the lines of "Adding message with id {} to Coda dataset {}"

**Review by as2388 — APPROVED** [review]
Thanks @DanielMwendwa

File: coda_v2_python_client/firebase_client_wrapper.py (hunk 2 above)
Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py:764]: Updated, thanks.

File: coda_v2_python_client/firebase_client_wrapper.py (hunk 1 above)
Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py:763]: Message cannot be accessed inside the add_in_transaction function. Running locals() inside functions returns all the arg in the parent function except for the message

Comment (DanielMwendwa) [reply coda_v2_python_client/firebase_client_wrapper.py:765]: Copied message outside the nested function

**Pull request merged**